python.d/puppet.chart.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+# Description: puppet netdata python.d module
+# Author: Andrey Galkin <andrey@futoin.org> (andvgal)
+#---
+# This module should work both with OpenSource and PE versions
+# of PuppetServer and PuppetDB.
+#
+# NOTE: PuppetDB may be configured to require proper TLS
+#       client certificate for security reasons. Use tls_key_file
+#       and tls_cert_file options then.
+#---
+
+from bases.FrameworkServices.UrlService import UrlService
+from json import loads
+import socket
+
+update_every = 5
+priority = 60000
+# very long clojure-based service startup time
+retries = 180
+
+ORDER = [
+    'jvm_heap',
+    'jvm_nonheap',
+    'cpu',
+    'fd_open',
+]
+CHARTS = {
+    'jvm_heap': {
+        'options': [None, "JVM Heap", "MB", "resources", "puppet.jvm", "area"],
+        'lines': [
+            ["jvm_heap_max", 'max', "absolute", 1, 1048576, 'hidden'],
+            ["jvm_heap_committed", 'committed', "absolute", 1, 1048576],
+            ["jvm_heap_used", 'used', "absolute", 1, 1048576],
+            ["jvm_heap_init", 'initial', "absolute", 1, 1048576, 'hidden'],
+        ]
+    },
+    'jvm_nonheap': {
+        'options': [None, "JVM Non-Heap", "MB", "resources", "puppet.jvm", "area"],
+        'lines': [
+            ["jvm_nonheap_max", 'max', "absolute", 1, 1048576, 'hidden'],
+            ["jvm_nonheap_committed", 'committed', "absolute", 1, 1048576],
+            ["jvm_nonheap_used", 'used', "absolute", 1, 1048576],
+            ["jvm_nonheap_init", 'initial', "absolute", 1, 1048576, 'hidden'],
```

Comment:
About all those `hidden`. You need them for alarms, right? Not a big deal but charts has [variables](https://github.com/firehol/netdata/wiki/External-Plugins#variable) for such cases (netdata doesn't Initializing db file for variable).

```
'jvm_nonheap': {
    'options': [None, "JVM Non-Heap", "MB", "resources", "puppet.jvm", "area"],
     'lines': [
        ["jvm_nonheap_committed", 'committed', "absolute", 1, 1048576],
    ]
    'variables': [
                ["jvm_nonheap_max"],
    ]
```

(yes, variable has no multiplier, divisor)

Reply:
Yeah, I've been considering variables, but understood the code wrong without digging further. Got initial impression that they are set only at chart creation time. Fixed.